lib/logstash/outputs/datadog_logs.rb

```diff
@@ -50,15 +50,19 @@ def close
   def multi_receive(events)
     return if events.empty?
     encoded_events = @codec.multi_encode(events)
-    if @use_http
-      batches = batch_http_events(encoded_events, DD_MAX_BATCH_LENGTH, DD_MAX_BATCH_SIZE)
-      batches.each do |batched_event|
-        process_encoded_payload(format_http_event_batch(batched_event))
-      end
-    else
-      encoded_events.each do |encoded_event|
-        process_encoded_payload(format_tcp_event(encoded_event.last, @api_key, DD_MAX_BATCH_SIZE))
+    begin
+      if @use_http
+        batches = batch_http_events(encoded_events, DD_MAX_BATCH_LENGTH, DD_MAX_BATCH_SIZE)
+        batches.each do |batched_event|
+          process_encoded_payload(format_http_event_batch(batched_event))
+        end
+      else
+        encoded_events.each do |encoded_event|
+          process_encoded_payload(format_tcp_event(encoded_event.last, @api_key, DD_MAX_BATCH_SIZE))
+        end
       end
+    rescue Exception => e
```

Comment:
`Exception` is the superclass of all exceptions, so it may be too broad and catch exceptions we wouldn't want to catch (syntax errors, signals, etc). We should probably catch `StandardError` instead?

```suggestion
    rescue => e
```

Reply:
You're right, fixing